processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/TypeSelector.java

```diff
@@ -149,6 +150,25 @@ else if ( pb.isTargetType() ) {
         }
     }
 
+    /**
+     * Adds default parameter bindings for the target-property-name if not already available
+     *
+     * @param availableParams Already available params, new entries will be added to this list
+     */
+    private <T extends Method> void addTargetPropertyNameBindings(List<ParameterBinding> availableParams) {
```

Comment:
Why not do what is done here in `addMappingTargetAndTargetTypeBindings`?

Reply:
Huh, I guess I was too narrow minded or too afraid of existing code changes.

I totally agree: additional method can be removed.

The question is will you do it, or shall I? 

If you prefer for me to do it, should I rename the private method to note all three actions i.e.: `addMappingTargetTargetTypeAndPropertyNameBindings` (:-S I don't like the name but the reality is such).

In any case If you want to change the code, I don't consider my consent necessary - since you're the expert.

Reply:
Feel free to change the code in any way it makes more sense to you!

Reply:
I agree with the naming `addMappingTargetAndTargetTypeAndPropertyNameBindings` being a bit too much, perhaps something like `addMissingAnnotationParamterBindings` would be shorter and still correct.

Reply:
I was only asking if there was any specific reason. Since there isn't, I'll adapt it. I've changed the name locally to `addTargetRelevantBindings`.